internal/backend/svc/svc.go

```diff
@@ -172,7 +172,13 @@ func makeFxOpts(cfg *Config, opts RunOptions) []fx.Option {
 		Component("projects", configset.Empty, fx.Provide(projects.New)),
 		Component("projectsgrpcsvc", projectsgrpcsvc.Configs, fx.Provide(projectsgrpcsvc.New)),
 		Component("envs", configset.Empty, fx.Provide(envs.New)),
-		Component("vars", configset.Empty, fx.Provide(vars.New)),
+		Component(
+			"vars",
+			configset.Empty,
+			fx.Provide(vars.New),
+			fx.Provide(func(v *vars.Vars) sdkservices.Vars { return v }),
+			fx.Invoke(func(v *vars.Vars, c sdkservices.Connections) { v.SetConnections(c) }),
```

Comment:
why don't you inject it as any other service  directly to vars.New ?

Reply:
to avoid cyclical dependency.

Reply:
which again, points to tight coupling between them which is bad

Reply:
agree.